pkg/proxy/iptables/proxier.go

```diff
@@ -495,7 +498,10 @@ func (proxier *Proxier) OnServiceUpdate(allServices []api.Service) {
 			}
 		}
 	}
-	proxier.syncProxyRules()
+	if expired := time.Since(proxier.lastSync); expired > proxier.minsyncPeriod {
```

Comment:
This isn't right.  If I am 10ms short of being able to refresh, I now have to wait for another event or for the longer sync-period.  Don't you want a timer to wake up in `proxier.minsyncPeriod - expired` ?